src/test/unit_tests/test_aws_sts.js

```diff
@@ -0,0 +1,65 @@
+/* Copyright (C) 2020 NooBaa */
+'use strict';
+
+const mocha = require('mocha');
+const assert = require('assert');
+const sinon = require('sinon');
+const AWS = require('aws-sdk');
+const cloud_utils = require('../../util/cloud_utils');
+const dbg = require('../../util/debug_module')(__filename);
+const fs = require("fs");
+
+mocha.describe('AWS STS tests', function() {
+    const projectedServiceAccountToken = "/var/run/secrets/openshift/serviceaccount/oidc-token";
+    const fakeAccessKeyId = "fakeAccessKeyId";
+    const fakeSecretAccessKey = "fakeSecretAccessKey";
+    const fakeSessionToken = "fakeSessionToken";
+    const defaultSTSCredsValidity = 3600;
+    const roleArn = "arn:aws:iam::261532230807:role/noobaa_s3_sts";
+    sinon.stub(fs.promises, "readFile").withArgs(projectedServiceAccountToken)
+        .returns("web-identity-token");
+        const stsFake = {
+            assumeRoleWithWebIdentity: sinon.stub().returnsThis(),
+            promise: sinon.stub().resolves({
+                    Credentials: {
+                        AccessKeyId: fakeAccessKeyId,
+                        SecretAccessKey: fakeSecretAccessKey,
+                        SessionToken: fakeSessionToken
+                    }
+            }),
+          };
+    const STSStub = sinon.stub(AWS, 'STS').callsFake(() => stsFake);
+
+    mocha.it('should generate aws sts creds', async function() {
+        const params = {
+            aws_sts_arn: roleArn
+        };
+        const expectedParams = [
+            { RoleArn: roleArn, RoleSessionName: 'testSession', WebIdentityToken: 'web-identity-token', DurationSeconds: defaultSTSCredsValidity }
+        ];
```

Comment:
we use the jsbeautifier settings - in vscode there's the HookyQR.beautify extension that will do it -
```suggest
        const expectedParams = [{
            RoleArn: roleArn,
            RoleSessionName: 'testSession',
            WebIdentityToken: 'web-identity-token',
            DurationSeconds: defaultSTSCredsValidity,
        }];
```